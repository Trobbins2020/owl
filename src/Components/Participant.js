import React from "react";

class Participant extends React.Component {
  render() {
    return (
      <div className="Participant">
        <p>Name: {this.props.name}</p>
        <p> inSession: {this.props.inSession ? "True" : "False"}</p>
        <img src={this.props.avatar} alt={this.props.avatar} />
        <h2>
          {this.props.timestamp !== undefined
            ? new Date(this.props.timestamp).toLocaleTimeString()
            : ""}
        </h2>
      </div>
    );
  }
}
export default Participant;
