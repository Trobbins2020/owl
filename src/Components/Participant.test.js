import Participant from "./Participant";
import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";

describe("Participant Component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Participant
        name="Talon"
        avatar={
          "https://scontent.fluh1-1.fna.fbcdn.net/v/t1.0-9/123766917_1253575681683191_9154823593112041728_o.jpg?_nc_cat=102&ccb=2&_nc_sid=09cbfe&_nc_ohc=OC0rGzbwf0IAX-AzQ5-&_nc_ht=scontent.fluh1-1.fna&oh=bbce5c965d82cc5b7e45c04400d08724&oe=5FD73F0D"
        }
        timestamp={1548852484247}
        inSession={true}
      />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer.create(<Participant />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
