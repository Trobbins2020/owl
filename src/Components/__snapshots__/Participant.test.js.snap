// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participant Component renders the UI as expected 1`] = `
<div
  className="Participant"
>
  <p>
    Name: 
  </p>
  <p>
     inSession: 
    False
  </p>
  <img />
  <h2>
    
  </h2>
</div>
`;
